src/neural/cuda/network_cuda.cc

```diff
@@ -416,8 +416,15 @@ class CudaNetwork : public Network {
       maxSize = std::max(maxSize, layer->GetOutputSize(max_batch_size_));
     }
 
-    if (use_res_block_winograd_fuse_opt_ && scratch_size_ > maxSize)
+
+    if ((attn_policy_ || use_res_block_winograd_fuse_opt_) && (scratch_size_ > maxSize)) {
+      // When attention is used, we split the tensor memory allocations into two
+      // halves and use them as scratch space to hold intermediate data
+      if (attn_policy_ && (scratch_size_ < maxSize * 2)) {
```

Comment:
another thing to consider is that attn_policy head isn't a single input output layer where intermediate steps have same size or less compared to input or output.  Especially with encoder blocks it is normal for inner steps to have larger size.  We may need to override GetMaxSize for attention policy head to determine the largest intermediate layer rather than just calculating the output size.

Reply:
hopefully fixed with latest commit. Can you please take a look.